pkg/orchestrator/handler/handler.go

```diff
@@ -0,0 +1,54 @@
+package handler
+
+import (
+	"github.com/go-logr/logr"
+	//"go.opencensus.io/plugin/ochttp"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+type orchestratorHandler struct {
+	log         logr.Logger
+	serviceFQDN string
+	transport   http.RoundTripper
+}
+
+func New(log logr.Logger, serviceFQDN string) http.Handler {
+	return &orchestratorHandler{
+		log:         log,
+		serviceFQDN: serviceFQDN,
+		//transport: AutoTransport,
+	}
+}
+
+// At present we just reverse proxy to the model but in future we will
+// make calls to other components as described by the Spec and combine
+// the responses into a single response.
+func (oh *orchestratorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	target := &url.URL{
+		Scheme: "http",
+		Host:   "istio-ingressgateway.istio-system.svc.cluster.local",
```

Comment:
Shouldn't we call the model directly rather than going back out through the istio ingress gateway?

Reply:
I wasn't sure at what level the knative plumbing kicks in. If we go directly to the Kubernetes Service I assume this won't bypass anything as knative just uses istio underneath so that will work as expected.

Reply:
Knative provides an istio virtual service that either points to activator or the service. As long as you hit that uri, you're good.

This value is available in the knservice.status.domain. We can also derive it, but that's a bit more challenging because it relies on a configmap.

I like that it's configurable, as the control plane can decide where the endpoint is.

In the control plane, perhaps we should simply wait to deploy the executor until the other components are initialized with their status.domains.

Reply:
yes here you would just want to point to knative service with the internal mesh gateway, the value on `status.domain` should work.

Reply:
Even better, we should use the derivable internal uri for it:
flowers-sample.default.svc.cluster.local
<name>.default.svc.cluster.local
```
kubectl get virtualservices.networking.istio.io -oyaml -n knative-serving
apiVersion: v1
items:
- apiVersion: networking.istio.io/v1alpha3
  kind: VirtualService
  metadata:
    annotations:
      networking.knative.dev/ingress.class: istio.ingress.networking.knative.dev
    creationTimestamp: 2019-05-29T03:34:53Z
    generation: 1
    labels:
      networking.internal.knative.dev/clusteringress: route-adfba6a8-81c2-11e9-8ddf-42010a8a0171
      serving.knative.dev/route: flowers-sample
      serving.knative.dev/routeNamespace: default
    name: route-adfba6a8-81c2-11e9-8ddf-42010a8a0171
    namespace: knative-serving
    ownerReferences:
    - apiVersion: networking.internal.knative.dev/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ClusterIngress
      name: route-adfba6a8-81c2-11e9-8ddf-42010a8a0171
      uid: b8bb9a1e-81c2-11e9-8ddf-42010a8a0171
    resourceVersion: "3013346"
    selfLink: /apis/networking.istio.io/v1alpha3/namespaces/knative-serving/virtualservices/route-adfba6a8-81c2-11e9-8ddf-42010a8a0171
    uid: b8bce2bb-81c2-11e9-8ddf-42010a8a0171
  spec:
    gateways:
    - knative-ingress-gateway
    - mesh
    hosts:
    - flowers-sample.default.example.com
    - flowers-sample.default.svc.cluster.local
    http:
    - appendHeaders:
        knative-serving-namespace: default
        knative-serving-revision: flowers-sample-default-9w2br
      match:
      - authority:
          regex: ^flowers-sample\.default(?::\d{1,5})?$
      - authority:
          regex: ^flowers-sample\.default\.example\.com(?::\d{1,5})?$
      - authority:
          regex: ^flowers-sample\.default\.svc(?::\d{1,5})?$
      - authority:
          regex: ^flowers-sample\.default\.svc\.cluster\.local(?::\d{1,5})?$
      retries:
        attempts: 3
        perTryTimeout: 10m0s
      route:
      - destination:
          host: activator-service.knative-serving.svc.cluster.local
          port:
            number: 80
        weight: 100
      timeout: 10m0s
      websocketUpgrade: true
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```